src/dialog-utilities.js

```diff
@@ -0,0 +1,39 @@
+import {Origin} from 'aurelia-metadata';
+
+/**
+* Returns a view model based on a given instruction
+*
+* @param {Object} instruction the instruction from which the view model is determined
+* @return {Promise} the view model
+*/
+export function getViewModel(instruction: any, compositionEngine: CompositionEngine) {
+  if (typeof instruction.viewModel === 'function') {
+    instruction.viewModel = Origin.get(instruction.viewModel).moduleId;
+  } else if (typeof instruction.viewModel === 'string') {
+    return compositionEngine.ensureViewModel(instruction);
+  }
+
+  return Promise.resolve(instruction);
+}
+
+/**
+* Invokes a lifecycle event on a given instance
+*
+* @param {Object} instance  the instance to invoke the event on
+* @param {String} name      the event to invoke
+* @param {Object} [model]   the optional model to invoke the event with
+* @return {Promise} the result of the invocation
+*/
+export function invokeLifecycle(instance: any, name: string, model: any) {
```

Comment:
`invokeLifecycle` now seems to have different logic so we need tests to cover the fix so we know the expected results.  ex - 

```
describe('.invokeLifecycle', () => {
  it('should return a resolved promise if blah blah', () => {
    let result = invokeLifecycle(new Promise());
    expect(result).toEqual(something);
  });
});
```